### PR TITLE
docs: establish i18n for message boxes

### LINKED ISSUE #41324: Translate contribute docs alerts
`docsify-plugin-flexible-alerts` lets us customise the alert labels, depending on the path.  This allows us to display hints in the appropriate language.

To use this, we'd need translations for each note (note, tip, warning and attention) and then we can configure the plugin.  Since we're translating into many languages, it might be best to handle this translation through Crowdin, even though it's only 4 words.

### PR DESCRIPTION
**Pull request by naomi-lgbt**
Checklist:

- [X] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [X] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [X] My pull request targets the `main` branch of freeCodeCamp.
- [X] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Closes #41324 

Sets up the i18n mapping for Chinese and Spanish for the message/alert boxes. Based on a discussion with Oliver, I think the best flow here is to:

- Add a language mapping when we are ready to download and serve the docs in that language
- Have a proofreader(s) provide the translations for that language 
- No need to upload on crowdin as these translations shouldn't change

### CODE REVIEW

**Comment by nhcarrigan** [comment]
@miyaliu666 and @RafaelDavisH - these are machine translations - could you verify when you have time?

**Comment by RafaelDavisH** [comment]
I added comments in the code or you prefer a PR to suggest changes?

File: docs/index.html [hunk 1]
```diff
@@ -89,9 +89,37 @@
...
       },
 
-      'flexible-alerts': {
-        style: 'callout'
-      },
+      // Add languages here for message box translations
+      'flexibleAlerts': {
+          note: {
+            label: {
+              '/i18n/chinese': '笔记',
```
Comment (ojeytonwilliams) [inline docs/index.html:96]:
```suggestion
              '/i18n/chinese/': '笔记',
```

File: docs/index.html [hunk 2]
```diff
@@ -89,9 +89,37 @@
...
+              '/i18n/espanol/': 'Nota',
+              '/': 'Note'
+            }
+          },
+          tip: {
+            label: {
+              '/i18n/chinese': '小费',
```
Comment (ojeytonwilliams) [inline docs/index.html:103]:
```suggestion
              '/i18n/chinese/': '小费',
```

File: docs/index.html [hunk 3]
```diff
@@ -89,9 +89,37 @@
...
+              '/i18n/espanol/': 'Tipo',
+              '/': 'Tip'
+            }
+          },
+          warning: {
+            label: {
+              '/i18n/chinese': '警告',
```
Comment (ojeytonwilliams) [inline docs/index.html:110]:
```suggestion
              '/i18n/chinese/': '警告',
```

File: docs/index.html [hunk 4]
```diff
@@ -89,9 +89,37 @@
...
+              '/i18n/espanol/': 'Aviso',
+              '/': 'Warning'
+            }
+          },
+          attention: {
+            label: {
+              '/i18n/chinese': '注意力',
```
Comment (ojeytonwilliams) [inline docs/index.html:117]:
```suggestion
              '/i18n/chinese/': '注意力',
```

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
I don't think the lack of `/` will matter, but it might as well be there.  Just in case.

**Comment by nhcarrigan** [comment]
> I added comments in the code or you prefer a PR to suggest changes?

Comments here is fine - I can apply them directly in a commit

**Review by ojeytonwilliams — DISMISSED** [review]
LGTM :+1: 

Once Miya and Rafa are happy, I think this can go in.  Then we can add other languages as needed (Portuguese next, since we have the docs already?).

File: docs/index.html [hunk 5]
```diff
@@ -89,9 +89,37 @@
...
       },
 
-      'flexible-alerts': {
-        style: 'callout'
-      },
+      // Add languages here for message box translations
+      'flexibleAlerts': {
+          note: {
+            label: {
+              '/i18n/chinese/': '笔记',
```
Comment (miyaliu666) [inline docs/index.html:96]: 注意

File: docs/index.html [hunk 6]
```diff
@@ -89,9 +89,37 @@
...
+              '/i18n/espanol/': 'Nota',
+              '/': 'Note'
+            }
+          },
+          tip: {
+            label: {
+              '/i18n/chinese/': '小费',
```
Comment (miyaliu666) [inline docs/index.html:103]: 提示

File: docs/index.html [hunk 7]
```diff
@@ -89,9 +89,37 @@
...
+          warning: {
+            label: {
+              '/i18n/chinese/': '警告',
+              '/i18n/espanol/': 'Aviso',
+              '/': 'Warning'
+            }
+          },
+          attention: {
+            label: {
+              '/i18n/chinese/': '注意力',
```
Comment (miyaliu666) [inline docs/index.html:117]: 注意

File: docs/index.html (hunk 5 above)
Comment (miyaliu666) [inline docs/index.html:96]: 注意

Comment (raisedadead) [inline docs/index.html:96]:
```suggestion
              '/i18n/chinese/': '注意',
```

Adding Miya's comment.

**Review by miyaliu666 — DISMISSED** [review]

**Comment by nhcarrigan** [comment]
Thanks @miyaliu666!

**Comment by nhcarrigan** [comment]
@RafaelDavisH if you could verify the Spanish translations are correct this will be ready to merge I think.

**Comment by nhcarrigan** [comment]
Thanks @RafaelDavisH!

**Comment by RandellDawson** [comment]
@ojeytonwilliams Is the person who will proofread the Portuguese different than the one who made most of the translations?  We would not want a proofreader to approve their own translations.

**Review by raisedadead — APPROVED** [review]

**Comment by raisedadead** [comment]
> Is the person who will proofread the Portuguese different than the one who made most of the translations? We would not want a proofreader to approve their own translations.

You are correct, we should encourage them to get at least one more person to help them.

**Pull request merged**